src/modules/account/hooks/useDeprecatedAccount.js

```diff
@@ -151,9 +152,28 @@ export const useDeprecatedAccount = () => {
         isMigrated: legacy?.data?.balance === '0',
         legacyAddress: legacy?.data?.legacyAddress,
       },
+      legacy: {
```

Comment:
If a few places including the `accountCard` component, we are null checking against `account.legacy`. Instead of optional chaining, the legacy property should be added conditionally.